src/tunnel.rs

```diff
@@ -413,31 +408,34 @@ make_tunnel_type! {
     NgrokLabeledTunnel, LabeledTunnel, label
 }
 
-pub async fn forward_tcp(id: &String, addr: String) -> PyResult<()> {
-    info!("Tunnel {id:?} TCP forwarding to {addr:?}");
-    let res = get_storage_by_id(id)
-        .await?
-        .tunnel
-        .lock()
-        .await
-        .fwd_tcp(addr)
-        .await;
-
-    debug!("forward_tcp returning");
-    canceled_is_ok(res)
-}
-
-pub async fn forward_pipe(id: &String, addr: String) -> PyResult<()> {
-    info!("Tunnel {id:?} Pipe forwarding to {addr:?}");
-    let res = get_storage_by_id(id)
-        .await?
-        .tunnel
-        .lock()
-        .await
-        .fwd_pipe(addr)
-        .await;
+pub async fn forward(id: &String, addr: String) -> PyResult<()> {
+    let tun = &get_storage_by_id(id).await?.tunnel;
+    let is_pipe =
+        addr.starts_with(PIPE_PREFIX) || addr.starts_with("unix:") || addr.starts_with("tun-");
+
+    let res = if is_pipe {
+        let mut tun_addr = addr.clone();
+        tun_addr = tun_addr
+            .strip_prefix(PIPE_PREFIX)
+            .unwrap_or(&tun_addr)
+            .to_string();
+        tun_addr = tun_addr
+            .strip_prefix("unix:")
+            .unwrap_or(&tun_addr)
+            .to_string();
+
+        info!("Tunnel {id:?} Pipe forwarding to {tun_addr:?}");
+        tun.lock().await.fwd_pipe(tun_addr).await
+    } else {
+        info!("Tunnel {id:?} TCP forwarding to {addr:?}");
+        tun.lock().await.fwd_tcp(addr).await
+    };
 
-    debug!("forward_pipe returning");
+    if is_pipe {
+        debug!("forward_pipe returning");
+    } else {
+        debug!("forward_tcp returning");
+    }
```

Comment:
Feel free to consolidate this down to one debug line, we'll know pipe vs tcp from the info logging above.